packages/insured-bridge-relayer/src/canonical-bridge-adapters/OptimismBridgeAdapter.ts

```diff
@@ -0,0 +1,90 @@
+import winston from "winston";
+import { ZERO_ADDRESS } from "@uma/common";
+
+import { providers } from "ethers";
+
+import { getMessagesAndProofsForL2Transaction } from "@eth-optimism/message-relayer";
+import { predeploys, getContractDefinition } from "@eth-optimism/contracts";
+import { getDeployedContractArtifact } from "@eth-optimism/contracts/dist/contract-deployed-artifacts";
+
+import BridgeAdapterInterface from "./BridgeAdapterInterface";
+
+import Web3 from "web3";
+import type { TransactionType } from "@uma/common";
+
+export class OptimismBridgeAdapter implements BridgeAdapterInterface {
+  private l1EthersProvider: any;
+  private l2EthersProvider: any;
+  private l1StateCommitmentChainAddress: any;
+  private l2CrossDomainMessengerAddress: any;
+  private l1CrossDomainMessenger: any;
+
+  constructor(readonly logger: winston.Logger, readonly l1Web3: Web3, readonly l2Web3: Web3) {}
+
+  async initialize() {
+    // Set providers for L1 and L2. Optimism package requires ethers providers.
+    this.l1EthersProvider = new providers.Web3Provider(this.l1Web3.currentProvider as any);
+    this.l2EthersProvider = new providers.Web3Provider(this.l2Web3.currentProvider as any);
+
+    // Fetch contract data.
+    this.l1StateCommitmentChainAddress = getDeployedContractArtifact("StateCommitmentChain", "mainnet").address;
+    this.l2CrossDomainMessengerAddress = predeploys.L2CrossDomainMessenger;
+    this.l1CrossDomainMessenger = new this.l1Web3.eth.Contract(
+      getContractDefinition("IL1CrossDomainMessenger").abi,
+      getDeployedContractArtifact("Proxy__OVM_L1CrossDomainMessenger", "mainnet").address
+    );
+  }
```

Comment:
Why do we need this method? Why can't we just put all of this in the constructor?

Or, if we just want to maintain the interface, you could just make this method empty.

It seems better to do as much as possible in the constructor since then you don't have to handle all the weird edge cases where stuff isn't defined.

Reply:
we need this method to make the interface conform with the `ArbitrumBridgeAdapter`, which needs this `initialize` method. they both need to conform to the `ExchangeAdapterInterface`.

you right though having it in the constructor is much cleaner. have moved it and set this to just `return`.